### PR TITLE
Constrain fx_stride order for scaled_mm

### PR DESCRIPTION
**Pull request by drisspg**
# Summary
CublasLT requires row_major @ col_major order for scaled_mm.  It is possible for the to not respect this constraint without adding this.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @yanbing-j @vkuzo @albanD

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -2216,7 +2216,7 @@ def fn(index):
 
 @register_lowering(aten.select_scatter, type_promotion_kind=None)
 def select_scatter(x, src, dim: int, index: int):
-    assert x.get_dtype() == src.get_dtype()
+    # assert x.get_dtype() == src.get_dtype()
```
Comment (eellison) [inline torch/_inductor/lowering.py:2219]: this is unrelated, right ?

**Review by eellison — APPROVED** [review]
looks good but remove unrelated comment out

File: torch/_inductor/lowering.py (hunk 1 above)
Comment (drisspg) [reply torch/_inductor/lowering.py:2219]: ohhh yup, sorry, unrelated to this one
